_posts/2015-09-05-automated-deployment.md

```diff
@@ -0,0 +1,30 @@
+---
+layout: post
+title: "Still stuck with FTP or Git pull to deploy? It's high time to embrace Automated Deployment with Capistrano"
+author: sumit_chhetri
+modified: 2015-09-05
+tags: [ deployment, devops, capistrano]
+---
+
+Are you still using FTP to deploy your code? Are you still manually performing the same steps of deploying a feature, again and again? How many hours have you spent on ssh-ing into the server, pulling the repo, migrating the database, reloading the web server and so on, for each deployment?? Ever wondered if there is a process as simple as a single click to perform all these steps for you?
+
+Automated Deployment does exactly these things for you. It takes the burden of remembering all the steps required in each deployment process and execute it smoothly.
+
+<!--more-->
+
+You don't need the rocket sicence to conclude that machine is reliable, efficient and faster than humans. In context to the software development, be it writing your code, testing your code and deploying your code, it is no different story. You need to leverage the tools and technologies available, if you want to be a better software engineer.
+
+In regard to software deployment, just try to count the number of times you deploy your software to staging, testing or production server, during its life cycle. I am pretty sure you don't have an exact number, or if you have, it is no less than hundreds. In each deployment you execute, it involves numerous steps; be it running some package installer scripts/commands, migrating your databases, notifying the clients about the deployment, and so on, which you will repeat again and again. You have no guarantee either, that the deployment will be successful each and everytime because YOU, a human, are deploying. Since you are deploying, it is more likely to be error prone than it would have been, had it been automated.
```

Comment:
In regards to software...

the project's life cycle


a human, is deploying.


more error prone than automated.